From 1fb986f3a722e8476af90b5ab512e495f0caf10f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristian=20H=C3=B8gsberg?= <krh@bitplanet.net>
Date: Thu, 6 Mar 2014 16:58:32 -0800
Subject: [PATCH] mesa: Don't optimize out glClear() if drawbuffer size is 0x0

The driver acquires buffer lazily and may not yet have buffers for
a newly current drawable.  In that case the drawable size is 0x0 but
we still need to call into the driver and make it fetch buffers and get
the drawable size.

https://bugs.freedesktop.org/show_bug.cgi?id=75797
---
 src/mesa/main/clear.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/mesa/main/clear.c b/src/mesa/main/clear.c
index 077c5fc..c3f7474 100644
--- a/src/mesa/main/clear.c
+++ b/src/mesa/main/clear.c
@@ -155,11 +155,6 @@ _mesa_Clear( GLbitfield mask )
       return;
    }
 
-   if (ctx->DrawBuffer->Width == 0 || ctx->DrawBuffer->Height == 0 ||
-       ctx->DrawBuffer->_Xmin >= ctx->DrawBuffer->_Xmax ||
-       ctx->DrawBuffer->_Ymin >= ctx->DrawBuffer->_Ymax)
-      return;
-
    if (ctx->RasterDiscard)
       return;
 
-- 
1.8.5.3

